### ISSUE TITLE
Make FANN C# work with .NET Core

### ISSUE THREAD

**Issue by joelself** [joelself/FannCSharp#3]
Someone contacted me privately and asked if I would be interested in making FANN C# work with .NET Core. I'm looking into seeing if can even be done, but if you would be interested in this, show your support here.

**Comment by joelself** [comment]
So it looks like the incompatibilities FANN C# has with .NET core are the `System.ApplicationException`, `System.SystemException` and `System.RuntimeInteropServices.HandleRe`f types. The exceptions are easy, we just throw different exceptions. The `HandleRef` is a little more complex, but it looks like there are other types like `GCHandle` that we can use, and/or we can just use `IntPtr`s and use `GC.KeepAlive`. I'm not entirely clear on `GCHandle` it seems like it can take an `IntPtr` or an object to wrap, but not both, so maybe you use `GCHandle` to hold the `IntPtr` and call `GC.KeepAlive` on the object holding the `IntPtr`? Or I could just copy the [`HandleRef` code from the Mono Project](https://github.com/mono/mono/blob/b7a308f660de8174b64697a422abfc7315d07b8c/mcs/class/corlib/System.Runtime.InteropServices/HandleRef.cs) (it looks pretty simple). I'm leaning towards copying Mono since I'm more confident that they know what they're doing than I'm confident that I would know how to translate `HandleRef` (the .NET API docs say `HandleRef` is more efficient anyways) use into use of other wrappers. Then I would have to decide if I want to change the `swig` source code to output different exceptions and calls to my custom `HandleRef` or do I just write clean up code that takes `swig` output and replaces .NET Core incompatible parts with compatible parts.

**Comment by ernestopedrosa** [comment]
Hi Joel. This is awesome. I can take a look on my side as well, but it seems you're far ahead on the process. Thanks.

**Comment by joelself** [comment]
I've forked (well, I can't fork my own repo, but made a new repo from this one) this repo into a [new one for .NET Core development](https://github.com/joelself/FANNCSharpCore). I created [an issue with some initial things I think I have to do to port existing projects over](https://github.com/joelself/FANNCSharpCore/issues/1). Since this "issue" no longer has to do with this project any more I'm going to close it and we can track the issue in the new "issue" on the new repo.
